feat: card Storybook Components

TenzDelek opened the pull request:
<!--   Thank you for your contribution. Before you submit the pull request:
1. Follow our contribution guidelines
2. Test your changes and attach their results to the pull request.
3. Update the relevant documentation.
-->

**Description**
created storybook for cards
1. Venue
<img width="1512" height="982" alt="Screenshot 2025-10-05 at 12 00 08 PM" src="https://github.com/user-attachments/assets/a85da663-daf1-4541-b598-22ccd6063f35" />

2. Speaker
<img width="1512" height="982" alt="Screenshot 2025-10-05 at 11 59 57 AM" src="https://github.com/user-attachments/assets/d6149d35-e4fc-4526-bfc8-d42ab2cb64f6" />

3. past-edition
<img width="1512" height="982" alt="Screenshot 2025-10-05 at 12 00 03 PM" src="https://github.com/user-attachments/assets/7ff4614a-d4d0-4382-82c4-3e2ef7e3caab" />


**Related issue(s)**
<!-- If you refer to a particular issue, provide its number, otherwise, remove this section.
For example, `Resolves #123`, `Fixes #43`, or `See also #33`. The `See also #33` option will not automatically close the issue after the PR merge. -->
GSOC

TenzDelek committed:
card

AceTheCreator committed:
Merge branch 'master' into card

AceTheCreator committed:
Merge branch 'master' into card

AceTheCreator reviewed (changes requested):
think we might need to rethink this card Storybook setup. The current approach treats stories as one-to-one with use cases, but they should really map one-to-one with components.

I understand your reasoning, since we have three card implementations, that’s three separate components, each with its own story file. But that separation should be because they’re technically different components, not because of different use cases.

A better approach would be to create a single, flexible Card component with variants that cover all three cases. The goal of a design system is to capture patterns and build reusable solutions. If every use case becomes its own component, we lose that consistency and end up with a collection of one-offs.

Great work so far 💪🏽
Let me know what you think, @TenzDelek.

cc @ashmit-coder

AceTheCreator committed:
Merge branch 'master' into card

TenzDelek committed:
Merge branch 'master' into card

TenzDelek commented:
> think we might need to rethink this card Storybook setup. The current approach treats stories as one-to-one with use cases, but they should really map one-to-one with components.
> 
> I understand your reasoning, since we have three card implementations, that’s three separate components, each with its own story file. But that separation should be because they’re technically different components, not because of different use cases.
> 
> A better approach would be to create a single, flexible Card component with variants that cover all three cases. The goal of a design system is to capture patterns and build reusable solutions. If every use case becomes its own component, we lose that consistency and end up with a collection of one-offs.
> 
> Great work so far 💪🏽 Let me know what you think, @TenzDelek.
> 
> cc @ashmit-coder

Thank you Ace for your suggestion but I think forcing those card into one component would be the wrong kind of abstraction and would be much harder to maintain in the long run. there will be unnecessary to much of prop passing and conditional.
I think instead of that, we can go with similar case of how shadcn does. using a base `card` component (kind of a wrapper).
cc @AceTheCreator  @ashmit-coder

TenzDelek committed:
Merge branch 'master' into card

TenzDelek committed:
Merge branch 'master' into card

TenzDelek commented:
hi @AceTheCreator ,@ashmit-coder  , I went through the codebase of the main website. and it seems they are also keeping it in a distributive way. I think this would be much approachable and cleaner than what you suggested. because thinking in another way all three card seems to be way different in terms of ui. but happy to hear what your thought on this one as well. 😄 
prev gsoc pr related with it https://github.com/asyncapi/website/pull/3122

TenzDelek committed:
Merge branch 'master' into card

AceTheCreator commented:
> hi @AceTheCreator ,@ashmit-coder , I went through the codebase of the main website. and it seems they are also keeping it in a distributive way. I think this would be much approachable and cleaner than what you suggested. because thinking in another way all three card seems to be way different in terms of ui. but happy to hear what your thought on this one as well. 😄 prev gsoc pr related with it [asyncapi/website#3122](https://github.com/asyncapi/website/pull/3122)

Okay, let's do it that way then 👍🏾

AceTheCreator committed:
Merge branch 'master' into card

AceTheCreator committed:
Merge branch 'master' into card

AceTheCreator commented on components/Venue/venue.stories.tsx:
```diff
@@ -0,0 +1,144 @@
+  cfp: null,
+  recordings: 'https://youtube.com/playlist?list=PLbi1gRlP7pijUwZJErzyYf_Rc-PWu4lXS',
+  playlist: 'https://youtube.com/playlist?list=PLbi1gRlP7pijUwZJErzyYf_Rc-PWu4lXS'
+};
```
I think we should only have `upcoming` `past` and `cfp` card type... Also, for the upcoming, i'd advise you use a date sometime in the future, else the `ended` tag is gonna keep showing. Just as seen below. 
<img width="836" height="1028" alt="Components-Card-VenueCard-CFP-Open-⋅-Storybook-12-12-2025_07_11_PM" src="https://github.com/user-attachments/assets/7e24d431-c07e-4279-822b-f0e971840d61" />

AceTheCreator commented on components/Speaker/speaker.stories.tsx:
```diff
@@ -0,0 +1,76 @@
+      <Speaker 
+        details={speakerDetails} 
+        location={hasLocation ? locationText : undefined}
+      />
```
Looking at this Speaker component got me thinking… why do we have a details prop and then location separately? Contextually, location should be part of the speaker’s details. Also, having a details prop feels too vague.

I think it would be cleaner if the Speaker component used more explicit, extracted props like:

<Speaker
  name={name}
  img={img}
  location={location}
  title={title}
/>


This makes the component’s props clearer and easier to understand at a glance.

AceTheCreator reviewed (commented):
@TenzDelek, are you planning on implementing the ticket card component?

TenzDelek committed:
filerestructure

TenzDelek committed:
Merge branch 'master' into card

TenzDelek committed:
Merge branch 'master' into card

TenzDelek committed:
some additional changes

TenzDelek committed:
Merge branch 'card' of github.com:TenzDelek/conference-website into card

TenzDelek committed:
build

TenzDelek committed:
Merge branch 'master' into card

TenzDelek committed:
remove unwanted

TenzDelek committed:
benifit update

TenzDelek commented:
ready for review @AceTheCreator

TenzDelek committed:
Merge branch 'master' into card

TenzDelek committed:
cleanup

AceTheCreator committed:
Merge branch 'master' into card

TenzDelek committed:
file naming

TenzDelek committed:
Merge branch 'card' of github.com:TenzDelek/conference-website into card

TenzDelek committed:
fix: correct Guidelines component filename case

Rename Guidelines.tsx to guidelines.tsx to fix case-sensitivity issue
in Linux build environments (Netlify). macOS is case-insensitive so the
issue wasn't caught locally, but the build was failing on CI.

AceTheCreator reviewed (approved):
LGTM! 🚀

TenzDelek committed:
Merge branch 'master' into card

TenzDelek committed:
Merge branch 'master' into card

TenzDelek committed:
Merge branch 'master' into card
